src/components/MultiHopTrade/components/LimitOrder/components/CancelLimtOrder.tsx

```diff
@@ -0,0 +1,222 @@
+import { useAppSelector, useSelectorWithArgs } from 'state/store'
+
+import { SwapperIcon } from '../../TradeInput/components/SwapperIcon/SwapperIcon'
+
+const cardBorderRadius = { base: '2xl' }
+
+export const CancelLimitOrder = () => {
+  const wallet = useWallet().state.wallet
+  const { setOrderToCancel } = useActions(limitOrderSlice.actions)
+  const { showErrorToast } = useErrorHandler()
+
+  const orderToCancel = useAppSelector(selectOrderToCancel)
+
+  const [cancelLimitOrders, { data: wasCancellationSuccessful, error, isLoading, reset }] =
+    useCancelLimitOrdersMutation()
+
+  useEffect(() => {
+    if (!error) return
+
+    const description = (error as CowSwapError).description ?? 'Unknown Error'
+
+    // TODO: Actually render a translated error description.
+    showErrorToast(description)
+  }, [error, showErrorToast])
+
+  useEffect(() => {
+    // Must explicitly check for false, because undefined means a cancellation isn't completed.
+    if (wasCancellationSuccessful !== false) return
+
+    showErrorToast(
+      `Failed to cancel order uid ${orderToCancel?.order.uid}`,
+      'limitOrder.cancel.cancellationFailed',
+    )
+  }, [error, orderToCancel?.order.uid, showErrorToast, wasCancellationSuccessful])
+
+  const handleClose = useCallback(() => {
+    reset()
+    setOrderToCancel(undefined)
+  }, [setOrderToCancel, reset])
+
+  const handleRequestCancellation = useCallback(async () => {
+    if (!orderToCancel || !wallet) {
+      return
+    }
+
+    await cancelLimitOrders({ wallet, ...orderToCancel })
+    setOrderToCancel(undefined)
+  }, [orderToCancel, wallet, cancelLimitOrders, setOrderToCancel])
+
+  const sellAsset = useSelectorWithArgs(selectAssetById, orderToCancel?.sellAssetId ?? '')
+  const buyAsset = useSelectorWithArgs(selectAssetById, orderToCancel?.buyAssetId ?? '')
+  const feeAsset = useSelectorWithArgs(selectFeeAssetById, orderToCancel?.sellAssetId ?? '')
+
+  const limitPrice = useMemo(() => {
+    if (!orderToCancel || !sellAsset || !buyAsset) return
+    const buyAmountCryptoPrecision = fromBaseUnit(orderToCancel.order.buyAmount, buyAsset.precision)
+    const sellAmountCryptoPrecision = fromBaseUnit(
+      orderToCancel.order.sellAmount,
+      sellAsset.precision,
+    )
+    return bn(buyAmountCryptoPrecision).div(sellAmountCryptoPrecision).toFixed()
+  }, [buyAsset, orderToCancel, sellAsset])
+
+  const expiryText = useMemo(() => {
+    const validTo = orderToCancel?.order.validTo
+    return validTo
+      ? formatDistanceToNow(validTo * 1000, {
+          addSuffix: true,
+        })
+      : undefined
+  }, [orderToCancel])
+
+  const formattedPercentage = useMemo(() => {
+    if (!orderToCancel) return '0'
+    const filledDecimalPercentage = bnOrZero(orderToCancel.order.executedSellAmount).div(
+      orderToCancel.order.sellAmount,
+    )
+    return filledDecimalPercentage.times(100).toFixed(2)
+  }, [orderToCancel])
+
+  return (
```

Comment:
ocd: early return `if (!orderToCancel)`

Reply:
I had this and it results in the modal instantly disappearing without animation (which looks janky), instead we use the presence to open and close the modal while keeping it mounted

Reply:
Sounds sane to me!